specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/virtualMachine.json

```diff
@@ -1891,6 +1891,75 @@
           "final-state-via": "location"
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/attachDetachDataDisks": {
+      "post": {
+        "tags": [
+          "VirtualMachines"
+        ],
+        "operationId": "VirtualMachines_AttachDetachDataDisks",
+        "description": "Attach and detach data disks to/from the virtual machine.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "vmName",
```

Comment:
>vmName [](http://example.com/codeflow?start=21&length=6)

Note - Creating a post action will not allow using  ARM Templates to attach/detach data disks.